extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/TracerRecorder.java

```diff
@@ -31,19 +35,38 @@
     }
 
     /* STATIC INIT */
-    public RuntimeValue<SdkTracerProvider> createTracerProvider(String serviceName, String serviceVersion,
+    public RuntimeValue<SdkTracerProvider> createTracerProvider(TracerConfig tracerConfig,
+            String serviceName,
+            String serviceVersion,
             ShutdownContext shutdownContext) {
-        BeanManager beanManager = Arc.container().beanManager();
+        ArcContainer container = Arc.container();
+        BeanManager beanManager = container.beanManager();
+
+        InjectableInstance<IdGenerator> idGenerator = container.select(IdGenerator.class);
+        InjectableInstance<Sampler> sampler = container.select(Sampler.class);
+
         SdkTracerProviderBuilder builder = SdkTracerProvider.builder();
 
+        // Define ID Generator if present
```

Comment:
@kenfinnigan changed it to a bean, and also added the sampler bean resolution so we can also set a custom sampler (Jaeger or other one) since they can require custom parameters for instantiation.

Reply:
Thanks for adding the sampling strategy as well

Reply:
Wondering whether this needs a multi faceted approach as well.

In pseudo logic:

1. Does a `Sampler` bean exist. If so, set it and done
2. Is there tracer config for sampling defined (based on the values in `Sampler`)? If so, use that sampling strategy.
3. Default sampler, ie. don't set anything on the builder.

What do you think? I think's important to enable simple sampling without needing to create a bean.

From a config perspective that might mean we can only support `always`, `off`, and `0.6`. It might also be possible to support parent based with a separate flag for it that defaults to false.

Reply:
I think it makes sense to have it mixed, going to add it and make it work like the auto configure options =]

Reply:
I've added the sampler config now in a way that if the bean exists it goes with the bean otherwise it will parse the properties, and also moved it to a runtime config so we can switch the sampler using env variables while rebooting the app so we don't need to rebuild the app to change a trace sampler ratio for instance =]